### PR TITLE
Team 6 map testing

### PR DESCRIPTION
**Pull request by TranHuuPhucAn**
# Description

Fixed a bug where item can spawn twice in a room, then added tests for GameArea, GameController, and MainRoom

Fixes / Closes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Adding tests for MainRoom 
- [x] Adding test for GameAreaTest
- [x] Adding for GameController 
- [x] further test coverage in LevelMap test
- [x] further test coverage in MainGameLevel

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: source/core/src/test/com/csse3200/game/areas/GameAreaTest.java [hunk 1]
```diff
@@ -1,8 +1,143 @@
...
+//    void testStopCurrentMusic() {
+//        when(mockResourceService.playMusic(anyString(), eq(true))).thenReturn(mockMusic);
+//
+//        gameArea.playMusic(GameArea.MusicType.NORMAL);
+//        gameArea.playMusic(GameArea.MusicType.BOSS);
+//        gameArea.stopCurrentMusic();
+//
+//        verify(mockMusic, times(2)).stop();
+//    }
+//
```
Comment (manya-k) [inline source/core/src/test/com/csse3200/game/areas/GameAreaTest.java:114]: why are they commented out?

File: source/core/src/main/com/csse3200/game/areas/GameController.java [hunk 2]
```diff
@@ -286,12 +280,13 @@ public void changeLevel(int levelNumber) {
...
-        minimap.addComponent(minimapComponent);
-        this.gameArea.spawnEntity(minimap);
+        gameArea.generateMinimap(getCurrentLevel());
+//        this.minimapFactory = new MinimapFactory(getCurrentLevel(), 0.5f);
+//        MinimapComponent minimapComponent = minimapFactory.createMinimap();
+//
+//        Entity minimap = new Entity();
+//        minimap.addComponent(minimapComponent);
+//        this.gameArea.spawnEntity(minimap);
     }
```
Comment (manya-k) [inline source/core/src/main/com/csse3200/game/areas/GameController.java:291]: why are they commented out?

**Review by manya-k — APPROVED** [review]
looks good overall

**Comment by manya-k** [comment]
pls resolve merge conflicts

**Review by Jas2310 — APPROVED** [review]

**Review by vsinh14 — APPROVED** [review]
LGTM all tests passing

**Pull request merged**